explainers/navigation-to-unsigned-bundles.md

```diff
@@ -0,0 +1,363 @@
+without signatures, to give users a format they can create from any website
+they're visiting, share it using existing peer-to-peer sharing apps,
+and then have a peer load it in their browser. The IETF has
+[approved](https://ietf.org/blog/ietf106-highlights/) but not yet created a
+WPACK working group to continue developing this format as of February 2020.
+
+Here's a [demo video](https://youtu.be/xAujz66la3Y) of this use-case for Web
+Bundles (a.k.a. Bundled HTTP Exchanges):
+
+<a href="https://www.youtube.com/watch?v=xAujz66la3Y">
+<img src="https://img.youtube.com/vi/xAujz66la3Y/0.jpg">
+</a>
+
+This explainer doesn't currently propose a standard way to identify all the
+resources from a loaded web page that need to be included in a bundle, or to
+identify a pre-packaged bundle that contains the current page. Instead it
+focuses on how to safely load and navigate within a bundle that's already
+created.
+
+### Relevant structure of a bundle
+
+A bundle is loaded from a "*bundle URL*", and it contains a set of HTTP *items*,
+each with a "*claimed URL*". The items are serialized in a particular order
+within the bundle, and this will affect the performance of loading the bundle
+when the bundle is loaded from a non-random-access medium (like the network),
+but it doesn't affect the semantics of the bundle.
+
+The bundle URL might be a `file://` URL containing private information like the
+user's name, which they might not want exposed to a website they saved for
+offline use.
+
+### Process of loading a bundle
+
+When the browser navigates to an `application/webbundle;v=___` resource, or
+opens a file that the filesystem identifies as that MIME type, it first parses
+the resource enough to pull out the bundle's [primary
+URL](https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#name-load-a-bundles-metadata)
+and a set of flags to control loading (not yet defined in the format). One of
+these flags defines whether the claimed URLs in the bundle are expected to be
+trusted, which for unsigned bundles means the bundle is served from the origin
+that claims it.
+
+### Loading a trusted unsigned bundle.
+
+The browser selects an **unsigned bundle scope** that's based on the bundle's
+URL in the same way the [service worker's scope
+restriction](https://w3c.github.io/ServiceWorker/#path-restriction) is based on
+the URL of the service worker script. If the bundle is served with the
+[`Service-Worker-Allowed`
+header](https://w3c.github.io/ServiceWorker/#service-worker-allowed), that sets
+the unsigned bundle scope to the value of that header. (We are also considering
+using a differently-named header that applies only to bundles.)
+
+Bundles can also contain a
+[manifest](https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#name-parsing-the-manifest-sectio)
+whose [`scope` key](https://www.w3.org/TR/appmanifest/#scope-member) can narrow
+the scope but not broaden it.
```

Comment:
I'm feeling inclined to drop this. PWAs having subresources out of its manifest's navigation scope is legitimate, but cannot be bundled as-is under this rule.

Probably the overhead of (waiting for the bytes + parsing) manifest is small, but since the internal redirect to the primary URL is on the critical path of bundle navigation I would like to keep it as fast as possible.